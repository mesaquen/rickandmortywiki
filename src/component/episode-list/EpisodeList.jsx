import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Skeleton from 'react-loading-skeleton';
import Text from '../text/Text';
/**
 * @author Mesaque Francisco <mesaquenf@gmail.com>
 * @since 2019-11-16
 */
const Container = styled.div`
  display: flex;
  flex-direction: column;
`;

const ListItem = styled.li`
  flex: 0 0 100%;
`;
const List = styled.ul`
  max-height: 200px;
  overflow-y: auto;
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  list-style-type: none;
  padding: 0;
`;
const Header = styled.div`
  margin-top: 12px;
`;
export default class EpisodeList extends Component {
  static propTypes = {
    episodes: PropTypes.arrayOf(PropTypes.object),
    ready: PropTypes.bool,
  };
  renderEpisodes = () => {
    const { episodes } = this.props;
    return episodes.map(episode => (
      <ListItem key={episode.id}>
        <Text size="12px">
          {episode.episode} - {episode.name}
        </Text>
      </ListItem>
    ));
  };

  renderSkeleton = () => {
    return (
      <ListItem>
        <Skeleton count={3} />
      </ListItem>
    );
  };
  render() {
    const { ready } = this.props;

    return (
      <>
        <Header>
          <Text size="20px" bold>
            Episodes
          </Text>
        </Header>
        <Container>
          <List>{ready ? this.renderEpisodes() : this.renderSkeleton()}</List>
        </Container>
      </>
    );
  }
}
