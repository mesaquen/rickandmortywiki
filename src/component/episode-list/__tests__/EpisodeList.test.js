import React from 'react';
import { render, cleanup } from '@testing-library/react';
import EpisodeList from '../EpisodeList';
/**
 * @author Mesaque Francisco <mesaquenf@gmail.com>
 * @since 2019-11-16
 */

describe('EpisodeList', () => {
  afterEach(cleanup);

  it('should have same snapshot when loading', () => {
    const { asFragment } = render(<EpisodeList ready={false} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should have same snapshot when rendering values', () => {
    const episodes = [
      { id: 1, episode: 'S01E01', name: 'one' },
      { id: 2, episode: 'S01E02', name: 'two' },
    ];

    const { asFragment } = render(<EpisodeList episodes={episodes} ready />);
    expect(asFragment()).toMatchSnapshot();
  });
});
