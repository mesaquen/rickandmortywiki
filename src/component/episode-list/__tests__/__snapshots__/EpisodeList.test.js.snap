// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodeList should have same snapshot when loading 1`] = `
<DocumentFragment>
  <div
    class="sc-chPdSV eErTkd"
  >
    <span
      class="sc-htpNat dwRXzK"
      color="#444"
    >
      Episodes
    </span>
  </div>
  <div
    class="sc-jzJRlG eFkJPq"
  >
    <ul
      class="sc-kAzzGY TbUrq"
    >
      <li
        class="sc-cSHVUG IZCuM"
      >
        <span>
          <span
            class="css-19bon8n react-loading-skeleton"
            style="animation: animation-16jpnkj 1.2s ease-in-out infinite;"
          >
            ‌
          </span>
          <span
            class="css-19bon8n react-loading-skeleton"
            style="animation: animation-16jpnkj 1.2s ease-in-out infinite;"
          >
            ‌
          </span>
          <span
            class="css-19bon8n react-loading-skeleton"
            style="animation: animation-16jpnkj 1.2s ease-in-out infinite;"
          >
            ‌
          </span>
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`EpisodeList should have same snapshot when rendering values 1`] = `
<DocumentFragment>
  <div
    class="sc-chPdSV eErTkd"
  >
    <span
      class="sc-htpNat dwRXzK"
      color="#444"
    >
      Episodes
    </span>
  </div>
  <div
    class="sc-jzJRlG eFkJPq"
  >
    <ul
      class="sc-kAzzGY TbUrq"
    >
      <li
        class="sc-cSHVUG IZCuM"
      >
        <span
          class="sc-htpNat cmzTFd"
          color="#444"
        >
          S01E01 - one
        </span>
      </li>
      <li
        class="sc-cSHVUG IZCuM"
      >
        <span
          class="sc-htpNat cmzTFd"
          color="#444"
        >
          S01E02 - two
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
